### PR TITLE
Use XDG_CACHE_HOME instead of XDG_CONFIG_CACHE

### PR DESCRIPTION
**Pull request by gangefors**
XDG_CONFIG_CACHE does not exist in the XDG Base directory listing.
XDG_CACHE_HOME is the correct env.var. to use for cache directories.

Reference: https://wiki.archlinux.org/title/XDG_Base_Directory
Fixes: #3008

### CODE REVIEW

File: src/ansiblelint/config.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 
 CACHE_DIR = (
-    os.path.expanduser(os.environ.get("XDG_CONFIG_CACHE", "~/.cache")) + "/ansible-lint"
+    os.path.expanduser(os.environ.get("XDG_CONFIG_HOME", "~/.cache")) + "/ansible-lint"
```
Comment (apatard) [inline src/ansiblelint/config.py:26]: What about $XDG_CACHE_HOME ?
Ref: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html:
"$XDG_CACHE_HOME defines the base directory relative to which user-specific non-essential data files should be stored. If $XDG_CACHE_HOME is either not set or empty, a default equal to $HOME/.cache should be used."

Comment (gangefors) [reply src/ansiblelint/config.py]: Yeah. I had it right in the first commit subject but managed to mess up all the rest.
Should be XDG_CACHE_HOME everywhere now.

**Review by ssbarnea — APPROVED** [review]

**Comment by ssbarnea** [comment]
Thanks for spotting this one!

**Review by apatard — APPROVED** [review]
thanks for noticing and fixing

**Pull request merged**